pam/deploy_template.yaml

```diff
@@ -0,0 +1,255 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: approval-pam
+objects:
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: approval-pam
+    labels:
+      app: approval
+    annotations:
+      description: All the KIE server web server's ports.
+  spec:
+    ports:
+    - name: approval-pam
+      port: 8080
+      targetPort: 8080
+    selector:
+      name: approval-pam
+- kind: Service
+  apiVersion: v1
+  spec:
+    clusterIP: "None"
+    ports:
+    - name: "ping"
+      port: 8888
+      targetPort: 8888
+    selector:
+      name: approval-pam
+  metadata:
+    name: approval-pam-ping
+    labels:
+      app: approval
+    annotations:
+      service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+      description: "The JGroups ping port for clustering."
+- kind: DeploymentConfig
+  apiVersion: v1
+  metadata:
+    name: approval-pam
+    labels:
+      app: approval
+    annotations:
+      template.alpha.openshift.io/wait-for-ready: "true"
+  spec:
+    strategy:
+      type: Recreate
+    triggers:
+    - type: ImageChange
+      imageChangeParams:
+        automatic: true
+        containerNames:
+        - approval-pam
+        from:
+          kind: ImageStreamTag
+          name: approval-pam:latest
+          namespace: ${IMAGE_NAMESPACE}
+    - type: ConfigChange
+    replicas: 1
+    selector:
+      name: approval-pam
+    template:
+      metadata:
+        name: approval-pam
+        labels:
+          app: approval
+          name: approval-pam
+      spec:
+        terminationGracePeriodSeconds: 60
+        containers:
+        - name: approval-pam
+          image: ${IMAGE_NAMESPACE}/approval-pam:latest
+          imagePullPolicy: Always
+          resources:
+            limits:
+              memory: "${KIE_SERVER_MEMORY_LIMIT}"
+          livenessProbe:
+            exec:
+              command:
+              - "/bin/bash"
+              - "-c"
+              - "echo 'curl --fail --silent -u $KIE_ADMIN_USER:$KIE_ADMIN_PWD http://localhost:8080/services/rest/server/healthcheck' > /tmp/live; bash /tmp/live"
```

Comment:
This is the bit that I thought was strange. What did the original implementation look like?

I would think a health check endpoint would not require authentication ...

Reply:
The original template has
```
curl --fail --silent -u '${KIE_ADMIN_USER}:${KIE_ADMIN_PWD}' http://localhost:8080/services/rest/server/healthcheck
```
The endpoint does require authentication. `${KIE_ADMIN_PWD}` was a parameter and got resolved into the true password at processing time. But now we no longer have this parameter.

Reply:
Shouldn't it still work as it was originally? Why add this file processing magic?

Reply:
Well with the env var being evaluated in the container rather than in the template